src/url.rs

```diff
@@ -79,6 +79,17 @@ pub fn parse_data_url(url: &Url) -> (String, String, Vec<u8>) {
     (media_type, charset, blob)
 }
 
+pub fn referer_url(url: Url) -> Url {
+    let mut url = url.clone();
+    // https://httpwg.org/specs/rfc9110.html#field.referer
+    // MUST NOT include the fragment and userinfo components of the URI
```

Comment:
"MUST NOT" sounds a bit too commanding, as if somebody's shouting at you.  "Must not" would be nicer and easier to read.